vcpkg/troubleshoot/asset-caching.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Troubleshoot asset caching issues
+description: Troubleshooting guide for common asset caching issues
+author: vicroms
+ms.author: viromer
+ms.date: 02/12/2024
+ms.topic: troubleshooting-general
+---
+
+# Troubleshoot asset caching issues
+
+This article is intended for users experiencing issues while setting
+up an asset cache. For instructions to set up an asset cache read the
+[asset caching documentation](../concepts/asset-caching.md).
+
+## Diagnosing asset caching errors
+
+When vcpkg downloads artifacts from the Internet (assets), it looks at
+available asset caches before attempting to download from the Internet.
+
+By default, vcpkg emits no warnings or errors when a package is not found on an
+asset cache. Adding [`x-block-origin`](../users/assetcaching.md#x-block-origin) to
+your asset cache configuration disables the silent download mechanism, turning
+failures to restore assets into package build failures.
+
+vcpkg prints the URL for each of the assets it downloads. You can detect asset
+caching errors if the download URL does not match your asset cache's URL. If
+you added `x-block-origin`, failing to restore an asset from one of the caches
+fails with an error message:
+
+```Console
+error: Failed to download from mirror set
+```
+
+## <a name="cache-failure"></a> Artifacts aren't being uploaded or restored from my asset cache
+
+The `x-azurl` asset cache backend provided by vcpkg, is designed to work with
+Azure Storage Containers, it may work with other storage services that accept
+PUT requests with simple token authentication.
+
+The configuration has the following format:
+
+`x-azurl,<url>,<sas>[,<rw>]`
+
+* `<url>`: The container's base URL
+* `<sas>`: A Shared Access Signature (SAS) token if working with Azure Storage
+  Containers; or an authentication request parameter if working with other providers.
+* `<rw>`: (Optional) read/write permission configuration
+
+> !NOTE
```

Comment:
```suggestion
> [!NOTE]
```

This is how it's currently rendering:
<img width="818" alt="image" src="https://github.com/microsoft/vcpkg-docs/assets/44559473/d25f8523-f53a-4c2f-a61d-f14b0c2262a5">